app.vue

```diff
@@ -24,27 +44,33 @@
         <v-btn @click="incrementX">
           Increment X
         </v-btn>
+        <v-checkbox
+          v-model="showComponents"
+          label="Show Vector Components"
+        ></v-checkbox>
       </v-container>
     </v-main>
   </v-app>
 </template>
 
 <script setup>
-import { ref } from 'vue'
+import { ref, computed } from 'vue'
```

Comment:
I thought these aren't needed if we're doing `<script setup>` but they _are_ needed if it's just `<script>`.

Reply:
I thought you needed computed, and Chat GPT said you did, FWIW. But when I remove it from the app.vue, it still works. 🤷‍♂️. Here's what Chat GPT says: 

You do need to import computed in script setup components in Vue 3 because computed is a part of Vue's Composition API, which isn’t globally available within the script setup context.

Unlike the Options API, where you can directly define computed properties as part of the component's options (like data or methods), the script setup syntax is more aligned with a modular approach. This requires you to explicitly import any Composition API functions you use, like ref, reactive, computed, and others.

The rationale behind this design choice is to:

Make it explicit: You know exactly what is being used from Vue's API and where it comes from, leading to better maintainability and a clearer understanding of dependencies.
Tree-shaking: Importing only the necessary functions allows tools like Webpack or Vite to tree-shake unused code, which reduces the final bundle size and improves performance.
This pattern of explicit imports is consistent with the way modern JavaScript modules work, promoting a more declarative and modular code structure.